### PR TITLE
Dj database url replacement

### PR DESCRIPTION
**Pull request by baruva**
In order to see us up for success in moving our Database related env vars we are changing the old way of getting the database URL which was mostly hard coding to a method that can take in. Currently this causes a failure as PGHOST is being set to a value that is not localhost not the actual host address when we create the pods but that seems to be on purpose and perhaps fixed in the other PR

---

## Additions

- adding env vars for Postgres standard env values like  PGUSER, PGPASSWORD, PGDATABASE, and PGHOST

## Removals

- Had coded values in the DATABASE_URL env var it now depends on the values supplied by the PG env vars now being added

## Changes

-

## How to test this PR

1.

## Notes and todos

-

## Checklist

- [ ] PR has an informative and human-readable title
  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [ ] Future todos are captured in comments and/or tickets
- [ ] Project documentation has been updated, potentially one or more of:
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance

#### Browser testing

Visually tested in the following supported browsers:
- [ ] Firefox
- [ ] Chrome
- [ ] Safari
- [ ] Edge 18 (the last Edge prior to it switching to Chromium)
- [ ] Internet Explorer 11 and 8 (via emulation in 11's dev tools)
- [ ] Safari on iOS
- [ ] Chrome on Android

#### Accessibility

- [ ] Keyboard friendly (navigable with tab, space, enter, arrow keys, etc.)
- [ ] Screen reader friendly
- [ ] Does not introduce new errors or warnings in [WAVE](https://wave.webaim.org/extension/)

#### Other

- [ ] Is useable without CSS
- [ ] Is useable without JS
- [ ] Does not introduce new lint warnings
- [ ] Flexible from small to large screens

### CODE REVIEW

File: .github/workflows/functional-tests.yml [hunk 1]
```diff
@@ -7,7 +7,11 @@ jobs:
   cypress:
     runs-on: ubuntu-latest
     env:
-      DATABASE_URL: postgres://cfpb:cfpb@localhost/cfgov
+      PGDATABASE: cfgov
+      PGUSER: cfpb
+      PGPASSWORD: cfpb
+      POSTGRES_USER: cfpb
+      PGHOST: localhost
       MAPBOX_ACCESS_TOKEN:  ${{ secrets.MAPBOX_ACCESS_TOKEN }}
```
Comment (jslay-excella) [inline .github/workflows/functional-tests.yml:15]: Missing `PGPORT`

File: cfgov/cfgov/settings/base.py [hunk 2]
```diff
@@ -237,9 +236,14 @@
...
-    "default": dj_database_url.config(
-        default="postgres://cfpb:cfpb@localhost/cfgov"
-    ),
+    "default": {
+        "ENGINE": "django.db.backends.postgresql",
+        "NAME": os.getenv("PGDATABASE", "cfgov"),
+        "USER": os.getenv("PGUSER", "cfpb"),
+        "PASSWORD": os.getenv("PGPASSWORD", "cfpb"),
+        "HOST": os.getenv("PGHOST", "localhost"),
+        "PORT": "5432",
```
Comment (jslay-excella) [inline cfgov/cfgov/settings/base.py:245]: This should use `PGPORT`

File: docker-compose.yml [hunk 3]
```diff
@@ -43,7 +43,12 @@ services:
         stdin_open: true
         tty: true
         environment:
-            DATABASE_URL: postgres://cfpb:cfpb@postgres/cfgov
+            PGDATABASE: cfgov
+            PGUSER: cfpb
+            PGPASSWORD: cfpb
+            POSTGRES_USER: cfpb
+            PGHOST: localhost
+            DATABASE_URL: postgres://cfpb:cfpb@localhost/cfgov
```
Comment (jslay-excella) [inline docker-compose.yml:51]: Missing `PGPORT`

File: docker-entrypoint.sh [hunk 4]
```diff
@@ -9,15 +9,15 @@ fi
 echo "Using $(python3 --version 2>&1) located at $(which python3)"
 
 # Wait for the database to be ready
-until psql "${DATABASE_URL}" -c '\q' >/dev/null 2>&1; do
+until psql "postgres://${PGUSER}:${PGPASSWORD}@${PGHOST}/${PGDATABASE}" -c '\q' >/dev/null 2>&1; do
```
Comment (jslay-excella) [inline docker-entrypoint.sh:12]: Missing `PGPORT`

File: docker-entrypoint.sh [hunk 5]
```diff
@@ -9,15 +9,15 @@ fi
...
   >&2 echo "Postgres is unavailable - waiting"
   sleep 1
 done
 
 # Do first-time set up of the database if necessary
 RUN_MIGRATIONS=${RUN_MIGRATIONS:-"true"}
 if [ "$RUN_MIGRATIONS" = true ]; then
-  if ! psql "${DATABASE_URL}" -c 'SELECT COUNT(*) FROM auth_user' >/dev/null 2>&1; then
+  if ! psql "postgres://${PGUSER}:${PGPASSWORD}@${PGHOST}/${PGDATABASE}" -c 'SELECT COUNT(*) FROM auth_user' >/dev/null 2>&1; then
```
Comment (jslay-excella) [inline docker-entrypoint.sh:20]: Missing `PGPORT`

File: docker-stack.yml [hunk 6]
```diff
@@ -61,7 +61,11 @@ services:
       ALLOW_ADMIN_URL: "True"
       APACHE_PORT: 8000
       APACHE_UPLOADS_F_ALIAS: /src/consumerfinance.gov/cfgov/f/
-      DATABASE_URL: postgres://cfpb@postgres/cfgov
+      PGDATABASE: cfgov
+      PGUSER: cfpb
+      PGPASSWORD: cfpb
+      POSTGRES_USER: cfpb
+      PGHOST: localhost
       DJANGO_ADMIN_USERNAME: admin
```
Comment (jslay-excella) [inline docker-stack.yml:69]: Missing `PGPORT`

File: dump-data.sh [hunk 7]
```diff
@@ -14,7 +14,7 @@ fi
 
 # If DATABASE_URL is not set, or is empty, default to localhost.
 if [ -z "$DATABASE_URL" ]; then
-    DATABASE_URL=postgres://cfpb:cfpb@localhost/cfgov
+    DATABASE_URL=postgres://$PGUSER:$PGPASSWORD@$PGHOST/$PGDATABASE
```
Comment (jslay-excella) [inline dump-data.sh:17]: Missing `PGPORT`

File: helm/cfgov/templates/deployment.yaml [hunk 8]
```diff
@@ -54,13 +54,27 @@ spec:
               value: {{ .Values.importDbPath | quote }}
             {{- end }}
             {{- if .Values.postgresql.enabled }}
+            - name: PGHOST
```
Comment (jslay-excella) [inline helm/cfgov/templates/deployment.yaml:57]: Please merge in my branch which replaces this with a template helper.

File: tox.ini [hunk 9]
```diff
@@ -77,7 +77,7 @@ commands=
 changedir=
     {toxinidir}/cfgov
 passenv=
-    DATABASE_URL ES_HOST ES_PORT GITHUB_ACTIONS SKIP_DJANGO_MIGRATIONS TEST_RUNNER
+    PGUSER PGPASSWORD PGDATABASE PGHOST ES_HOST ES_PORT GITHUB_ACTIONS SKIP_DJANGO_MIGRATIONS TEST_RUNNER
```
Comment (jslay-excella) [inline tox.ini:80]: Missing `PGPORT`

File: tox.ini [hunk 10]
```diff
@@ -141,11 +141,15 @@ changedir={[unittest-config]changedir}
 deps=
     {[unittest-config]deps}
 passenv=
-    DATABASE_URL
+    PGUSER PGPASSWORD PGDATABASE PGHOST
```
Comment (jslay-excella) [inline tox.ini:144]: Missing `PGPORT`

File: tox.ini [hunk 11]
```diff
@@ -141,11 +141,15 @@ changedir={[unittest-config]changedir}
...
 setenv=
+    PGUSER={env:PGUSER:cfpb
+    PGPASSWORD={env:PGPASSWORD:cfpb}
+    PGDATABASE={env:PGDATABASE:cfgov}
+    PGHOST={env:PGHOST:localhost}
     DJANGO_SETTINGS_MODULE=cfgov.settings.production
```
Comment (jslay-excella) [inline tox.ini:150]: Missing `PGPORT`

File: tox.ini [hunk 12]
```diff
@@ -141,11 +141,15 @@ changedir={[unittest-config]changedir}
...
     DJANGO_STATIC_ROOT={toxinidir}/collectstatic
-    DATABASE_URL={env:DATABASE_URL:postgres://cfpb:cfpb@localhost/cfgov}
+    DATABASE_URL=postgres://{env:PGUSER:cfpb}:{env:PGPASSWORD:cfpb}@{env:PGHOST:localhost}/{env:PGDATABASE:cfgov}
```
Comment (jslay-excella) [inline tox.ini:152]: Missing `PGPORT`

**Review by jslay-excella — CHANGES_REQUESTED** [review]
* Need to merge in my Postgres Update branch to bring in the new helper template. 
* Add missing `PGPORT` with default of `5432`. I know we rarely, if ever, define differently, but still should be there.

**Comment by jslay-excella** [comment]
Also, why is a brand new branch starting out with merge commits from 2 months ago?

image

**Comment by willbarton** [comment]
Just a quick note before I give a more thorough review: we cannot merge these changes without corresponding changes to our deployments, etc, otherwise they'll break.

**Pull request closed without merging**